Nodejs/Product/Nodejs/Project/ImportWizard/ImportSettings.cs

```diff
@@ -114,101 +125,110 @@ private static void RecalculateIsValid(DependencyObject d, DependencyPropertyCha
+        private static void SourceOrLanguage_Updated(DependencyObject d, DependencyPropertyChangedEventArgs e)
         {
             if (!d.Dispatcher.CheckAccess())
             {
-                d.Dispatcher.BeginInvoke((Action)(() => SourcePath_Updated(d, e)));
+                d.Dispatcher.BeginInvoke((Action)(() => SourceOrLanguage_Updated(d, e)));
                 return;
             }
 
             RecalculateIsValid(d, e);
 
-            var s = d as ImportSettings;
-            if (s == null)
+            var settings = d as ImportSettings;
+            if (settings == null)
             {
                 return;
             }
 
-            if (string.IsNullOrEmpty(s.ProjectPath) || s._isAutoGeneratedProjectPath)
+            if (string.IsNullOrEmpty(settings.ProjectPath) || settings.isAutoGeneratedProjectPath)
             {
-                s.ProjectPath = CommonUtils.GetAvailableFilename(s.SourcePath, Path.GetFileName(s.SourcePath), ".njsproj");
-                s._isAutoGeneratedProjectPath = true;
+                settings.ProjectPath = CommonUtils.GetAvailableFilename(settings.SourcePath, Path.GetFileName(settings.SourcePath), ".njsproj");
+                settings.isAutoGeneratedProjectPath = true;
             }
 
-            var sourcePath = s.SourcePath;
+            var sourcePath = settings.SourcePath;
             if (Directory.Exists(sourcePath))
             {
-                var filters = s.Filters;
-                var dispatcher = s.Dispatcher;
+                // cache so we don't get a cross thread exception when accessing the settings object
+                var filters = settings.Filters;
+                var dispatcher = settings.Dispatcher;
+                var language = settings.ProjectLanguage;
 
                 // Nice async machinery does not work correctly in unit-tests,
                 // so using Dispatcher directly.
                 Task.Factory.StartNew(() =>
                 {
-                    var files = Directory.EnumerateFiles(sourcePath, "*.js", SearchOption.TopDirectoryOnly);
+                    string[] fileList = null;
 
-                    if (filters.Split(';').Any(x => x.EndsWith(NodejsConstants.TypeScriptExtension, StringComparison.OrdinalIgnoreCase)))
+                    // find a logical startup file
+                    // ignore tsx and jsx for now
+                    switch (language)
                     {
-                        files = Directory.EnumerateFiles(
-                            sourcePath,
-                            "*.ts",
-                            SearchOption.TopDirectoryOnly
-                        ).Concat(files);
+                        case ProjectLanguage.TypeScript:
+                            fileList = EnumerateTopLevelFiles("*.ts");
+                            break;
+                        case ProjectLanguage.JavaScript:
+                        default:
```

Comment:
>default [](start = 24, length = 7)

Assert that we never hit the default case?

Reply:
yep